pandas/_libs/groupby.pyx

```diff
@@ -93,6 +124,13 @@ cdef inline float64_t median_linear(float64_t* a, int n) nogil:
         a = tmp
         n -= na_count
 
+    return calc_median_linear(a, n, na_count)
```

Comment:
Could you free `a` here instead of in `calc_median_linear`? While it does mean you'll have to free in both this method and the masked version, it makes `calc_median_linear` have no side-effects.

Reply:
Moved it, my train of thought went the other way, if moved inside the function the caller would not forget it. But no real preference on my side